testsuite/elytron/src/test/java/org/wildfly/test/security/common/elytron/SecurityDomainLockReleaseTest.java

```diff
@@ -0,0 +1,92 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.test.security.common.elytron;
+
+import jakarta.inject.Inject;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildFlyRunner;
+
+import java.util.concurrent.TimeUnit;
+
+@RunWith(WildFlyRunner.class)
+public class SecurityDomainLockReleaseTest {
+    CLIWrapper cli;
+
+    @Inject
+    private static ServerController serverController;
+
+    @Before
+    public void setup() throws Exception {
+        cli = new CLIWrapper(true);
+
+        cli.sendLine("/subsystem=elytron/key-store=exampleKeystore:add(path=keystore, relative-to=jboss.server.config.dir, type=JKS, credential-reference={clear-text=secret})", true);
+        cli.sendLine("/subsystem=elytron/key-store=exampleKeystore:generate-key-pair(alias=localhost,algorithm=RSA,key-size=1024,validity=365,distinguished-name=\"CN=localhost\")", true);
+        cli.sendLine("/subsystem=elytron/key-store=exampleKeystore:store()", true);
+        cli.sendLine("/subsystem=elytron/filesystem-realm=exampleSecurityRealm:add(path=fs-realm-users,relative-to=jboss.server.config.dir, key-store=exampleKeystore, key-store-alias=localhost)", true);
+        cli.sendLine("/subsystem=elytron/filesystem-realm=exampleSecurityRealm:add-identity(identity=user1)", true);
+        cli.sendLine("/subsystem=elytron/filesystem-realm=exampleSecurityRealm:set-password(identity=user1, clear={password=\"passwordUser1\"})", true);
+        cli.sendLine("/subsystem=elytron/filesystem-realm=exampleSecurityRealm:add-identity-attribute(identity=user1, name=Roles, value=[\"Admin\",\"Guest\"])", true);
+        cli.sendLine("/subsystem=elytron/security-domain=exampleSecurityDomain:add(default-realm=exampleSecurityRealm,permission-mapper=default-permission-mapper,realms=[{realm=exampleSecurityRealm}])", true);
+        cli.sendLine("reload");
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        removeTestResources();
+        cli.close();
+    }
+
+    @Test
+    public void testReadIdentityReleasingLock() throws Exception {
+        boolean success = cli.sendLine("/subsystem=elytron/security-domain=exampleSecurityDomain:read-identity(name=user1)", true);
+        Assert.assertTrue(success);
+
+        Thread updateThread = new Thread(() -> {
+            cli.sendLine("/subsystem=elytron/filesystem-realm=exampleSecurityRealm:update-key-pair()", true);
+        });
+        updateThread.start();
+
+        boolean deadlock = false;
```

Comment:
Thanks for adding this test case @lvydra!

Your analysis is indeed correct, the `read-identity` operation was acquiring the shared lock for the identity but wasn't releasing it, resulting in a hang because the `update-key-pair` operation needs the exclusive lock in order to proceed.

Now that we've confirmed that this test case resulted in a hang before your changes and passes after your changes, I think we can update the test body here to just assert that the `update-key-pair` operation executes successfully.

Might be good to also try calling `read-identity` again after the call to `update-key-pair`.